holoviews/core/data.py

```diff
@@ -469,6 +464,11 @@ def validate(cls, columns):
 
 
     @classmethod
+    def check_dense(cls, arrays):
+        return any(array.shape not in [arrays[0].shape, (1,)] for array in arrays[1:])
```

Comment:
If I understand this code correctly ``check_compressed`` might be a better name...